Analysis/Tasks/PWGDQ/dileptonEE.cxx

```diff
@@ -320,20 +336,124 @@ struct DileptonEE {
+  Filter filterTrackSelected = aod::reducedtrack::isBarrelSelected > uint8_t(0);
+
+  void init(o2::framework::InitContext&)
+  {
+    fValues = new float[VarManager::kNVars];
+    VarManager::SetDefaultVarNames();
+    fHistMan = new HistogramManager("analysisHistos", "aa", VarManager::kNVars);
+    fHistMan->SetUseDefaultVariableNames(kTRUE);
+    fHistMan->SetDefaultVarNames(VarManager::fgVariableNames, VarManager::fgVariableUnits);
+
+    TString histNames = "";
+    TString configCutNamesStr = fConfigElectronCuts.value;
+    if (!configCutNamesStr.IsNull()) {
+      std::unique_ptr<TObjArray> objArray(configCutNamesStr.Tokenize(","));
+      for (int icut = 0; icut < objArray->GetEntries(); ++icut) {
+        fCutNames.push_back(objArray->At(icut)->GetName());
+        histNames += Form("PairsBarrelMEULS_%s;PairsBarrelMELSpp_%s;PairsBarrelMELSnn_%s;", fCutNames[icut].Data(), fCutNames[icut].Data(), fCutNames[icut].Data());
+        fTwoTrackFilterMask |= (uint8_t(1) << icut);
+      }
+    }
+
+    DefineHistograms(fHistMan, histNames.Data());    // define all histograms
+    VarManager::SetUseVars(fHistMan->GetUsedVars()); // provide the list of required variables so that VarManager knows what to fill
+    fOutputList.setObject(fHistMan->GetMainHistogramList());
+  }
+
+  void process(soa::Filtered<MyEventsHashSelected>& events, soa::Filtered<MyBarrelTracksSelected> const& tracks)
+  {
+    uint8_t twoTrackFilter = 0;
+
+    events.bindExternalIndices(&tracks);
+    auto tracksTuple = std::make_tuple(tracks);
+    AnalysisDataProcessorBuilder::GroupSlicer slicerTracks(events, tracksTuple);
+
+    // Strictly upper categorised collisions, for 100 combinations per bin, skipping those in entry -1
+    for (auto& [event1, event2] : selfCombinations("fMixingHash", 10, -1, events, events)) {
+
+      // event informaiton is required to fill histograms where both event and pair information is required (e.g. inv.mass vs centrality)
+      VarManager::ResetValues(0, VarManager::kNVars, fValues);
+      VarManager::FillEvent<gkEventFillMap>(event1, fValues);
+
+      auto it1 = slicerTracks.begin();
+      auto it2 = slicerTracks.begin();
+      for (auto& slice : slicerTracks) {
+        if (slice.groupingElement().index() == event1.index()) {
+          it1 = slice;
+          break;
+        }
+      }
+      for (auto& slice : slicerTracks) {
+        if (slice.groupingElement().index() == event2.index()) {
+          it2 = slice;
+          break;
+        }
+      }
+
+      auto tracks1 = std::get<soa::Filtered<MyBarrelTracksSelected>>(it1.associatedTables());
+      tracks1.bindExternalIndices(&events);
+      auto tracks2 = std::get<soa::Filtered<MyBarrelTracksSelected>>(it2.associatedTables());
+      tracks2.bindExternalIndices(&events);
+
+      for (auto& track1 : tracks1) {
+        for (auto& track2 : tracks2) {
+          twoTrackFilter = track1.isBarrelSelected() & track2.isBarrelSelected() & fTwoTrackFilterMask;
```

Comment:
the mask can be taken out as it has no effect

Reply:
done